From b2f290e25b9825b86ec5e536bfde964ddf61e0de Mon Sep 17 00:00:00 2001
From: "Patil, Rachna" <rachna@ti.com>
Date: Wed, 29 Aug 2012 13:33:58 +0530
Subject: [PATCH 1065/1092] ARM: OMAP: AM33XX: Add platform data for TSC/ADC
 MFD

This patch adds platform data related to registering
TSC/ADC as an MFDevice.

Signed-off-by: Patil, Rachna <rachna@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 17 +++++++++++------
 arch/arm/mach-omap2/devices.c         | 10 +++++-----
 arch/arm/mach-omap2/devices.h         |  2 +-
 3 files changed, 17 insertions(+), 12 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 801d339..758538d 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -37,6 +37,7 @@
 #include <linux/mfd/tps65217.h>
 #include <linux/pwm_backlight.h>
 #include <linux/input/ti_tsc.h>
+#include <linux/mfd/ti_tscadc.h>
 #include <linux/reboot.h>
 #include <linux/pwm/pwm.h>
 #include <linux/opp.h>
@@ -153,6 +154,10 @@ static struct tsc_data am335x_touchscreen_data  = {
 	.steps_to_configure = 5,
 };
 
+static struct mfd_tscadc_board tscadc = {
+	.tsc_init = &am335x_touchscreen_data,
+};
+
 static u8 am335x_iis_serializer_direction1[] = {
 	INACTIVE_MODE,	INACTIVE_MODE,	TX_MODE,	RX_MODE,
 	INACTIVE_MODE,	INACTIVE_MODE,	INACTIVE_MODE,	INACTIVE_MODE,
@@ -444,7 +449,7 @@ static struct pinmux_config lcdc_pin_mux[] = {
 	{NULL, 0},
 };
 
-static struct pinmux_config tsc_pin_mux[] = {
+static struct pinmux_config tscadc_pin_mux[] = {
 	{"ain0.ain0",           OMAP_MUX_MODE0 | AM33XX_INPUT_EN},
 	{"ain1.ain1",           OMAP_MUX_MODE0 | AM33XX_INPUT_EN},
 	{"ain2.ain2",           OMAP_MUX_MODE0 | AM33XX_INPUT_EN},
@@ -1067,12 +1072,12 @@ static void lcdc_init(int evm_id, int profile)
 	return;
 }
 
-static void tsc_init(int evm_id, int profile)
+static void mfd_tscadc_init(int evm_id, int profile)
 {
 	int err;
 
-	setup_pin_mux(tsc_pin_mux);
-	err = am33xx_register_tsc(&am335x_touchscreen_data);
+	setup_pin_mux(tscadc_pin_mux);
+	err = am33xx_register_mfd_tscadc(&tscadc);
 	if (err)
 		pr_err("failed to register touchscreen device\n");
 }
@@ -1966,7 +1971,7 @@ static struct evm_dev_cfg gen_purp_evm_dev_cfg[] = {
 						PROFILE_2 | PROFILE_7) },
 	{lcdc_init,	DEV_ON_DGHTR_BRD, (PROFILE_0 | PROFILE_1 |
 						PROFILE_2 | PROFILE_7) },
-	{tsc_init,	DEV_ON_DGHTR_BRD, (PROFILE_0 | PROFILE_1 |
+	{mfd_tscadc_init,	DEV_ON_DGHTR_BRD, (PROFILE_0 | PROFILE_1 |
 						PROFILE_2 | PROFILE_7) },
 	{rgmii1_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
 	{rgmii2_init,	DEV_ON_DGHTR_BRD, (PROFILE_1 | PROFILE_2 |
@@ -2038,7 +2043,7 @@ static struct evm_dev_cfg evm_sk_dev_cfg[] = {
 	{rgmii2_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
 	{lcdc_init,     DEV_ON_BASEBOARD, PROFILE_ALL},
 	{enable_ecap2,     DEV_ON_BASEBOARD, PROFILE_ALL},
-	{tsc_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
+	{mfd_tscadc_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
 	{gpio_keys_init,  DEV_ON_BASEBOARD, PROFILE_ALL},
 	{gpio_led_init,  DEV_ON_BASEBOARD, PROFILE_ALL},
 	{lis331dlh_init, DEV_ON_BASEBOARD, PROFILE_ALL},
diff --git a/arch/arm/mach-omap2/devices.c b/arch/arm/mach-omap2/devices.c
index a42791e..6898a2e 100644
--- a/arch/arm/mach-omap2/devices.c
+++ b/arch/arm/mach-omap2/devices.c
@@ -25,7 +25,7 @@
 #include <linux/can/platform/d_can.h>
 #include <linux/platform_data/uio_pruss.h>
 #include <linux/pwm/pwm.h>
-#include <linux/input/ti_tsc.h>
+#include <linux/mfd/ti_tscadc.h>
 
 #include <mach/hardware.h>
 #include <mach/irqs.h>
@@ -174,22 +174,22 @@ int __init am33xx_register_lcdc(struct da8xx_lcdc_platform_data *pdata)
 	return 0;
 }
 
-int __init am33xx_register_tsc(struct tsc_data *pdata)
+int __init am33xx_register_mfd_tscadc(struct mfd_tscadc_board *pdata)
 {
 	int id = -1;
 	struct platform_device *pdev;
 	struct omap_hwmod *oh;
 	char *oh_name = "adc_tsc";
-	char *dev_name = "tsc";
+	char *dev_name = "ti_tscadc";
 
 	oh = omap_hwmod_lookup(oh_name);
 	if (!oh) {
-		pr_err("Could not look up TSC%d hwmod\n", id);
+		pr_err("Could not look up TSCADC%d hwmod\n", id);
 		return -ENODEV;
 	}
 
 	pdev = omap_device_build(dev_name, id, oh, pdata,
-			sizeof(struct tsc_data), NULL, 0, 0);
+			sizeof(struct mfd_tscadc_board), NULL, 0, 0);
 
 	WARN(IS_ERR(pdev), "Can't build omap_device for %s:%s.\n",
 			dev_name, oh->name);
diff --git a/arch/arm/mach-omap2/devices.h b/arch/arm/mach-omap2/devices.h
index 4725325..65dd439 100644
--- a/arch/arm/mach-omap2/devices.h
+++ b/arch/arm/mach-omap2/devices.h
@@ -17,7 +17,7 @@ struct isp_platform_data;
 int omap3_init_camera(struct isp_platform_data *pdata);
 
 int __init am335x_register_mcasp(struct snd_platform_data *pdata, int ctrl_nr);
-extern int __init am33xx_register_tsc(struct tsc_data *pdata);
+extern int __init am33xx_register_mfd_tscadc(struct mfd_tscadc_board *pdata);
 extern int __init am33xx_register_ecap(int id,
 		struct pwmss_platform_data *pdata);
 extern int __init am33xx_register_ehrpwm(int id,
-- 
1.7.11.2

